metagraph/src/annotation/binary_matrix/base/binary_matrix.cpp

```diff
@@ -32,6 +34,25 @@ BinaryMatrix::slice_rows(const std::vector<Row> &row_ids) const {
     return slice;
 }
 
+void BinaryMatrix::slice_columns(const std::vector<Column> &column_ids,
+                                 const ColumnCallback &callback) const {
+    size_t nrows = num_rows();
+
+    #pragma omp taskloop
+    for (size_t k = 0; k < column_ids.size(); ++k) {
+        Column j = column_ids[k];
+        auto column = get_column(j);
+        size_t nsetbits = column.size();
+        callback(j, bit_vector_smart([s=std::move(column)](const auto &index_callback) {
+            for (uint64_t i : s) {
+                index_callback(i);
+            }
+        }, nrows, nsetbits));
+    }
+
+    #pragma omp taskwait
```

Comment:
Using more threads will use more RAM as well, which is not cool. Maybe do it with a single thread?

Reply:
good point, maybe we should just have a safe single-threaded implementation here and leave it up to the other methods to implement a multi-threaded one if there's a nice way to do it

Reply:
Yes, `get_column` is exposed, so you can always write this parallel loop in the caller.